packages/ec-core/src/operator/selector/lexicase.rs

```diff
@@ -95,3 +100,200 @@ where
             .context("The pool of candidates was empty")
     }
 }
+
+#[cfg(test)]
+#[rustversion::attr(before(1.81), allow(clippy::unwrap_used))]
+#[rustversion::attr(
+    since(1.81),
+    expect(
+        clippy::unwrap_used,
+        reason = "Panicking is the best way to deal with errors in unit tests"
+    )
+)]
+mod tests {
+    use std::iter::once;
+
+    use proptest::{
+        collection,
+        prelude::{any, Strategy},
+        prop_assert, prop_assert_eq,
+    };
+    use test_strategy::proptest;
+
+    use super::*;
+    use crate::individual::ec::EcIndividual;
+
+    // Generate a population from a vector of single scores, one per individual.
+    // These have to be converted to iterators using `once` so they're "like"
+    // arrays of scores.
+    fn population_from_single_scores<I>(scores: I) -> Vec<EcIndividual<usize, TestResults<i32>>>
+    where
+        I: IntoIterator<Item = i32>,
```

Comment:
```suggestion
    fn population_from_single_scores(scores: impl IntoIterator<Item = i32>) -> Vec<EcIndividual<usize, TestResults<i32>>>
```

this could be simplified like so :)